### ISSUE TITLE
suspected_diffs action should provide report_id, matching urls, % match, and number of words

### ISSUE THREAD

**Issue by ragesoss** [valhallasw/plagiabot#2]
Right now, if I query for suspected diffs of an article (like [this](http://tools.wmflabs.org/eranbot/plagiabot/api.py?action=suspected_diffs&page_title=AucShun)) I don't get enough information to investigate the suspected plagiarism.

**Comment by eranroz** [comment]
added support in the API to show the result in raw form from the DB.

**Comment by kaldari** [comment]
We would like to be able to automatically pull the URLs from plagiabot into the Copyvio Detector tool on Tool Labs. Unfortunately, this information is currently mixed in with other information and HTML in the API results. Would it be possible to have the API output this data as an independent data node (i.e. not mixed with other data)?

**Comment by fhocutt** [comment]
The web API outputs data as stored in the database, and the reports are stored as one blob. It would be reasonably straightforward to parse them to return the information separately, but would also be pretty hacky. On the other hand, storing each independently would require changes to the database schema, and more extensive changes to the code. Thoughts, @eranroz, @kaldari?

ETA: on another look, I see that this gets even more complicated when there are multiple parts with different associated ithenticate reports. It's probably easiest for this project to figure out how to return that data in the API reasonably cleanly and not fuss with the database.

**Comment by eranroz** [comment]
Yes, this isn't pretty DB - this is just a hack and all the report is stored as a one large blob. It is possible to store specific relevant fields outside this BLOB in some less messy way, but I don't think it's a good idea to store everything. maybe a new suspected_urls table with columns: copyright_diff_id, URL, similarity?

**Comment by fhocutt** [comment]
Ah, missed this comment. The basic thing we would like to do is have enough information for someone using Copyvio Detector to be able to identify and follow up on something that ithenticate has flagged: https://phabricator.wikimedia.org/T110144 . 

In the short term I'll just add an API method to parse the report blob, but if you think it's worth modifying the DB, it would be useful to be able to access everything relevant that goes into the report -- a separate table, definitely, to account for multiple suspected sources per report. Would copyright_diff_id be the same as the report ID? It would be useful to have % and number of words for similarity as well.

Also, incidentally, is there a dev branch set up somewhere, or are there any instructions or suggestions on setting up a dev environment? We have a community-tech-tools project on Tool Labs, but we don't have the ithenticate creds (though a copy of part of the DB would do as well for this issue, come to think of it), and I haven't set up a Pywikibot-based tool like this one before.
